runtime-sdk/modules/evm/src/lib.rs

```diff
@@ -197,104 +327,139 @@ impl API for Module {
     }
 }
 
-impl Module {
-    const EVM_CONFIG: Config = Config::istanbul();
+impl<Cfg: Config> Module<Cfg> {
+    const EVM_CONFIG: EVMConfig = EVMConfig::istanbul();
 
-    fn do_evm<C, F, V>(
-        source: H160,
-        value: U256,
-        gas_price_in: U256,
-        gas_limit: u64,
-        ctx: &mut C,
-        f: F,
-    ) -> Result<V, Error>
+    fn do_evm<C, F, V>(source: H160, value: U256, ctx: &mut C, f: F) -> Result<V, Error>
     where
         F: FnOnce(
             &mut StackExecutor<'static, MemoryStackState<'_, 'static, evm_backend::Backend<'_, C>>>,
+            u64,
         ) -> (evm::ExitReason, V),
-        C: Context,
+        C: TxContext,
     {
-        // Make sure that gas_price >= min_gas_price.
         let params = Self::params(ctx.runtime_state());
-        let gas_price: primitive_types::U256 = gas_price_in.into();
+        let den = params.token_denomination;
         let min_gas_price: primitive_types::U256 = params.min_gas_price.into();
+
+        let gas_limit: u64 = core::Module::remaining_tx_gas(ctx);
+        let gas_price: primitive_types::U256 = ctx.tx_auth_info().fee.gas_price().into();
+
+        // Make sure that gas_price >= min_gas_price.
```

Comment:
We could handle this at the core module level and optionally enforce a min gas price for all SDK transactions, not only EVM (since now the EVM gas price is basically the same as the SDK gas price)? In that case it could be part of core parameters and be handled in `before_handle_call` of the core module. Thoughts?

Reply:
That's a good idea!  I'll modify the code.